homeassistant/components/reolink/media_source.py

```diff
@@ -0,0 +1,321 @@
+        )
+        for status in statuses:
+            for day in status.days:
+                children.append(
+                    BrowseMediaSource(
+                        domain=DOMAIN,
+                        identifier=f"DAY+{config_entry_id}+{channel}+{stream}+{status.year}+{status.month}+{day}",
+                        media_class=MediaClass.DIRECTORY,
+                        media_content_type=MediaType.PLAYLIST,
+                        title=f"{status.year}/{status.month}/{day}",
+                        can_play=False,
+                        can_expand=True,
+                    )
+                )
+
+        res_name = "High res." if stream == "main" else "Low res."
+        return BrowseMediaSource(
+            domain=DOMAIN,
+            identifier=f"DAYS+{config_entry_id}+{channel}+{stream}",
+            media_class=MediaClass.CHANNEL,
+            media_content_type=MediaType.PLAYLIST,
+            title=f"{host.api.camera_name(channel)} {res_name}",
+            can_play=False,
+            can_expand=True,
+            children=children,
+        )
+
+    async def _generate_camera_files(
+        self,
+        config_entry_id: str,
+        channel: int,
+        stream: str,
+        year: int,
+        month: int,
+        day: int,
+    ) -> BrowseMediaSource:
+        """Return all recording files on a specific day of a Reolink camera."""
+        host = self.data[config_entry_id].host
+
+        start = dt.datetime(year, month, day, hour=0, minute=0, second=0)
+        end = dt.datetime(year, month, day, hour=23, minute=59, second=59)
+
+        children = []
+        _LOGGER.debug(
+            "Requesting VODs of %s on %s/%s/%s",
+            host.api.camera_name(channel),
+            year,
+            month,
+            day,
+        )
+        (_, vod_files) = await host.api.request_vod_files(
+            channel, start, end, stream=stream
+        )
+        for file in vod_files:
+            file_name = f"{file.start_time.time()} {file.duration}"
+            if file.triggers != file.triggers.NONE:
+                file_name += " " + " ".join(
+                    str(trigger.name).title()
+                    for trigger in file.triggers
+                    if trigger != trigger.NONE
+                )
+
+            children.append(
+                BrowseMediaSource(
+                    domain=DOMAIN,
+                    identifier=f"FILE+{config_entry_id}+{channel}+{stream}+{file.file_name}",
+                    media_class=MediaClass.VIDEO,
+                    media_content_type=MediaType.VIDEO,
+                    title=file_name,
+                    can_play=True,
+                    can_expand=False,
+                )
+            )
+
+        res_name = "High res." if stream == "main" else "Low res."
+        return BrowseMediaSource(
+            domain=DOMAIN,
+            identifier=f"FILES+{config_entry_id}+{channel}+{stream}",
```

Comment:
I'd pick something besides `+` as a separator since `+` tends to have issues with url encoding

Reply:
What seperator would you suggest?
Maybe "|"?

Reply:
`|` should be much safer